### PR TITLE
widgets: Add display_unit to KymotrackerParameter

### PR DESCRIPTION
**Pull request by mikhas**
Also make sure that we get proper display units for spatially aware parameter by quering the kymo instance.

### CODE REVIEW

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py [hunk 1]
```diff
@@ -847,6 +851,7 @@ def _get_default_parameters(kymo, channel):
             r"tracking noise. Note that this length refers to the number of detected points, not "
             r"length in time!",
             abridged_name="Min length",
+            display_unit="pts",
```
Comment (JoepVanlier) [inline lumicks/pylake/nb_widgets/kymotracker_widgets.py:854]: Why not write it out in full?

File: lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py [hunk 2]
```diff
@@ -385,9 +385,9 @@ def test_keyword_args(kymograph):
     "gain,line_time,pixel_size,ref_values",
     (
         # fmt:off
-        (1, 2.0, 5.0, {"pixel_threshold": (97, 1, 99), "track_width": (4 * 5, 3 * 5, 15 * 5), "sigma": (2 * 5, 1 * 5, 5 * 5), "velocity": (0, -5 * 5/2, 5 * 5/2)}),
-        (0, 2.0, 5.0, {"pixel_threshold": (1, 1, 2), "track_width": (4 * 5, 3 * 5, 15 * 5), "sigma": (2 * 5, 1 * 5, 5 * 5), "velocity": (0, -5 * 5/2, 5 * 5/2)}),
-        (1, 4.0, 4.0, {"pixel_threshold": (97, 1, 99), "track_width": (4 * 4, 3 * 4, 15 * 4), "sigma": (2 * 4, 1 * 4, 5 * 4), "velocity": (0, -5 * 4/4, 5 * 4/4)}),
+        (1, 2.0, 5.0, {"pixel_threshold": (97, 1, 99,"counts"), "track_width": (4 * 5, 3 * 5, 15 * 5,"$\\mu$m"), "sigma": (2 * 5, 1 * 5, 5 * 5,"$\\mu$m"), "velocity": (0, -5 * 5/2, 5 * 5/2,"$\\mu$m/s")}),
```
Comment (JoepVanlier) [inline lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py:388]: nitpick: formatting (also applies to the others)

```suggestion
        (1, 2.0, 5.0, {"pixel_threshold": (97, 1, 99, "counts"), "track_width": (4 * 5, 3 * 5, 15 * 5, "$\\mu$m"), "sigma": (2 * 5, 1 * 5, 5 * 5, "$\\mu$m"), "velocity": (0, -5 * 5/2, 5 * 5/2, "$\\mu$m/s")}),
```

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py [hunk 3]
```diff
@@ -884,6 +891,7 @@ def _get_default_parameters(kymo, channel):
             r"When tracking, the algorithm searches for points in future frames to connect. Points "
             r"within a certain distance from the expected future position are connected. The "
             r"diffusion parameter determines how quickly this distance grows over time.",
+            display_unit=f"{kymo._calibration.unit_label}²/s",
```
Comment (JoepVanlier) [inline lumicks/pylake/nb_widgets/kymotracker_widgets.py:894]: One thing that strikes me as odd is that `unit_label` is TeX style, while here we have a unicode squared symbol. Not really an issue, but curious.

That said, do we want this label as TeX?

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Main question I have is whether we want this field as TeX.

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py (hunk 1 above)
Comment (mikhas) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py:854]: IKR? Ongoing discussion over here...

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py (hunk 3 above)
Comment (mikhas) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py:894]: Either the whole unit becomes a $..$ math section, or none of it, IMO.

Comment (JoepVanlier) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py:894]: I agree. Personally, I'm leaning towards not having the unit be a math section, but just having the mu symbol in `Kymo`'s unit label.

Comment (mikhas) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py:893]: I've decided to leave that issue for a separate PR, OK?

File: lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py [hunk 4]
```diff
@@ -403,10 +403,11 @@ def test_default_params_img_dependent(gain, line_time, pixel_size, ref_values):
 
     default_params = _get_default_parameters(kymo, "red")
     for key, param in ref_values.items():
-        value, mini, maxi = param
+        value, mini, maxi, dui = param
```
[the remark this answers is not in the document]
Comment (mikhas) [reply lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py]: done

File: lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py (hunk 2 above)
Comment (mikhas) [reply lumicks/pylake/nb_widgets/tests/test_kymotracker_widget.py]: added spaces

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py (hunk 1 above)
Comment (mikhas) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py]: done

File: lumicks/pylake/nb_widgets/kymotracker_widgets.py (hunk 3 above)
Comment (JoepVanlier) [reply lumicks/pylake/nb_widgets/kymotracker_widgets.py:893]: Sure.

**Pull request merged**